### DISCUSSION TITLE
Using cypress with uniforms...

### DISCUSSION THREAD

**Discussion by mdesousa** [vazco/uniforms#1050] [Q&A] [answered]
Hi, has anyone had any experience using cypress.io (or similar ui testing frameworks) with uniforms?
We are looking for the best way to get reference to dom elements. It appears that uniforms is generating ids like `uniforms-0000-0002`. Do we know if these ids are constant for each field? How are these IDs generated? Would they be a reliable way to get field references for testing purposes?

Thanks!

**Comment by radekmie** [comment, marked as answer]
Hi @mdesousa! The `id`s are generated in a stable way, but I wouldn't recommend relying on them for tests. You can do two things:
1. Provide your own `id`s or any other prop like `data-cy` and use them in tests.
2. Rely on `name`s instead. These are not randomized at all and have a business meaning.
